### PR TITLE
(SIMP-3834) Fix hierarchy to permit `poss` + `none`

### PR DESCRIPTION
**Pull request by op-ct**
The patch fixes an issue where the `poss` scenario could not be used
because Hiera unconditionally loaded extra SIMP classes, particualrly on
unsupported RedHat-family systems (e.g., OracleLinux)

SIMP-3834 #close
SIMP-3832 #close

### CODE REVIEW

**Comment by op-ct** [comment]
This PR requires https://github.com/simp/pupmod-simp-pupmod/pull/71

File: hiera.yaml [hunk 1]
```diff
@@ -2,15 +2,21 @@
...
+  - name: "OS + Release"
+    backend: "yaml"
+    path: "os/%{facts.operatingsystem}-%{facts.operatingsystemmajrelease}"
+  - name: "OS"
+    backend: "yaml"
+    path: "os/%{facts.operatingsystem}"
   - name: "OSFamily + Release"
     backend: "yaml"
-    path: "os/%{facts.osfamily}-%{facts.operatingsystemmajrelease}"
+    path: "os/family/%{facts.osfamily}-%{facts.operatingsystemmajrelease}"
```
Comment (trevor-vaughan) [inline hiera.yaml]: This doesn't make sense. Families don't have releases.

File: .fixtures.yml [hunk 2]
```diff
@@ -22,27 +24,40 @@ fixtures:
...
+      repo: https://github.com/hercules-team/augeasproviders_syslog
+      ref: 2.1.1
     autofs: https://github.com/simp/pupmod-simp-autofs
     chkrootkit: https://github.com/simp/pupmod-simp-chkrootkit
     clamav: https://github.com/simp/pupmod-simp-clamav
     concat:
       # master is at 4.0.1, but we are bound to < 4.0.0 in our metadata.json
       repo: https://github.com/simp/puppetlabs-concat
-      ref: 3.0.0
+      ref: 2.2.1
```
Comment (jeefberkey) [inline .fixtures.yml]: We have moved to 3.0.0, this is not an accurate test

File: hiera.yaml (hunk 1 above)
Comment (op-ct) [reply hiera.yaml]: @trevor-vaughan I can't speak for whoever originally designed this hierarchy, but it makes perfect sense to me: we routinely speak of "EL7" and "EL6" with the understanding that they are  `:operatingsystem`-independent releases that are consistent across the entire `RedHat` OS family (e.g., CentOS-7 and RedHat-7).

**Comment by lamawithonel** [comment]
I'm not apposed to doing multiple things in a single PR, but given our current policy of squash-and-merge, this needs to be slimmed down. I'm seeing at least three different changes here: one for the hiera stuff, one for `assert_metadata()`, and one to add API to `spec_helper.rb`. And that's just on a cursory look.

File: .fixtures.yml (hunk 2 above)
Comment (op-ct) [reply .fixtures.yml]: @jeefberkey I noticed that, and then I noticed that 3.0.0 didn't resolve with many of the other fixtures (some augeasproviders IIRC constrained it to 2.x).  Sadly, we continue to use unconstrained  tracking branches in `.fixtures.yml`, and many of these repositories have recently made major releases that we _also_ don't support.

Unfortunately, the HEAD of simp-core's `Puppetfile.stable` hadn't been updated enough to  sort them out unambiguously, so I resolved what I could from this .fixtures file.  If you can suggest an alternate set of versioned tags that are more correct (and resolve), I'd certainly welcome a patch.

File: manifests/admin.pp [hunk 3]
```diff
@@ -62,6 +62,8 @@
   Boolean          $set_polkit_admin_group    = true
 ){
 
+  simplib::assert_metadata( $module_name )
```
Comment (jeefberkey) [inline manifests/admin.pp]: The reintroduction of these functions at least merits a Y bump. This change may cause unexpected failures when using a version of this module greater than `~> 1`, even if the platform is unsupported by SIMP.

**Comment by jeefberkey** [comment]
This PR was merged without proper review. Every update to the PR commands 2 approvals before it can be accepted.

**Pull request merged**